tools/deviceGen/input/device_features.json

```diff
@@ -0,0 +1,2666 @@
+            "value": true,
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderStorageImageArrayDynamicIndexing",
+            "comment": "[EXPOSE AS A LIMIT] ROADMAP 2022 but Apple GPUs have poor support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderClipDistance",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] good device support"
+        },
+        {
+            "type": "bool",
+            "name": "shaderCullDistance",
+            "value": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderFloat64",
+            "comment": "[EXPOSE AS A LIMIT] Cannot be always enabled cause Intel ARC is handicapped"
+        },
+        {
+            "type": "bool",
+            "name": "shaderInt64",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE]"
+        },
+        {
+            "type": "bool",
+            "name": "shaderInt16",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2024"
+        },
+        {
+            "groupComment": [
+                "poor support on Apple GPUs"
+            ]
+        },
+        {
+            "type": "bool",
+            "name": "shaderResourceResidency",
+            "value": false
+        },
+        {
+            "type": "bool",
+            "name": "shaderResourceMinLod",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "[TODO LATER] once we implemented sparse resources"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseBinding",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidencyBuffer",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidencyImage2D",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidencyImage3D",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidency2Samples",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidency4Samples",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidency8Samples",
+            "expose": false
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "sparseResidency16Samples",
```

Comment:
should be in the above group, in `sparseResidencyAliased`

Reply:
Done